controllers/machine_controller_phases.go

```diff
@@ -220,7 +220,7 @@ func (r *MachineReconciler) reconcileInfrastructure(ctx context.Context, m *clus
 		return err
 	}
 
-	if m.Status.InfrastructureReady || !infraConfig.GetDeletionTimestamp().IsZero() {
+	if !infraConfig.GetDeletionTimestamp().IsZero() {
```

Comment:
We should also set the readiness status immediately after calling `external.IsReady`, specifically `m.Status.InfrastructureReady = true` should become `m.Status.InfrastructureReady = ready`

Reply:
made the discussed changes, please take a look. Thx